### PR TITLE
EREGCSC-314 Create SubpartReaderView

### PR DESCRIPTION
**Pull request by cgodwin1**

### CODE REVIEW

File: regulations/views/mixins.py [hunk 1]
```diff
@@ -14,6 +14,8 @@ def build_citation(context):
             citation.append(context["part"])
         if 'section' in context:
             citation.append(context["section"])
+        if 'subpart' in context:
+            citation.append(context["subpart"])
```
Comment (whytheplatypus) [inline regulations/views/mixins.py]: At the moment a `citation` doesn't contain both a section and subpart. Add some logic here to make model that.

**Review by whytheplatypus — CHANGES_REQUESTED** [review]

File: regulations/views/mixins.py [hunk 2]
```diff
@@ -53,15 +56,39 @@ def build_sidebar_context(self, sidebar_class, label_id, version):
...
+
+        if view_name == "subpart_reader_view" and subpart is not None:
+            view = view_name
+            args = (part, subpart, version)
+        elif view_name == "part_reader_view":
+            view = view_name
+            args = (part, version)
+        else:
+            view = self.default_view
+            args = (part, section, version)
```
Comment (whytheplatypus) [inline regulations/views/mixins.py]: This conditional logic can move into the individual view classes now via an override of `get_url`

File: regulations/views/reader.py [hunk 3]
```diff
@@ -24,19 +25,19 @@ def get_context_data(self, **kwargs):
         reg_version = context["version"]
         reg_part = context["part"]
         reg_citation = context["citation"]
-        toc = self.get_toc(reg_part, reg_version)
+        toc = self.get_toc(context['part'], context['version'])
```
Comment (whytheplatypus) [inline regulations/views/reader.py]: Can this continue to use the `reg_part` and `reg_citation` variables?

File: regulations/views/reader.py [hunk 4]
```diff
@@ -59,3 +60,61 @@ def get_neighboring_sections(self, label, version):
...
+
+        part = context['part']
+        version = context['version']
+        first_section = utils.first_section(part, version)
+        first_subpart = utils.first_subpart(part, version)
+
+        c = {
+            'subpart_view_link': reverse('subpart_reader_view', args=(part, first_subpart, version)),
+            'section_view_link': reverse('section_reader_view', args=(part, first_section, version)),
+        }
```
Comment (whytheplatypus) [inline regulations/views/reader.py]: move this into a function with a common name across the classes.

**Review by whytheplatypus — APPROVED** [review]

**Review by whytheplatypus — CHANGES_REQUESTED** [review]
clicked the wrong button :facepalm:

File: regulations/views/reader.py (hunk 4 above)
Comment (cgodwin1) [reply regulations/views/reader.py]: Implemented in c8537d2

File: regulations/views/reader.py (hunk 3 above)
Comment (cgodwin1) [reply regulations/views/reader.py]: Resolved in a2e2a69

File: regulations/views/mixins.py (hunk 2 above)
Comment (cgodwin1) [reply regulations/views/mixins.py]: Implemented in 5f17599

File: regulations/views/mixins.py (hunk 1 above)
Comment (cgodwin1) [reply regulations/views/mixins.py]: Resolved in d865569

File: regulations/views/mixins.py [hunk 5]
```diff
@@ -2,18 +2,23 @@
...
 from regulations.generator.toc import fetch_toc
-from regulations.generator.section_url import SectionUrl
 
 
 def build_citation(context):
         citation = []
         if 'part' in context:
             citation.append(context["part"])
+        else:
+            return []
```
Comment (whytheplatypus) [inline regulations/views/mixins.py]: I messed this up. The function shouldn't return two different types, so this should return an empty string or the following conditionals should be nested inside this if e.g.

```python
        if 'part' in context:
            citation.append(context["part"])
            if 'section' in context:
                citation.append(context["section"])
            elif 'subpart' in context:
                citation.append(context["subpart"])
        return "-".join(citation)
```

**Review by whytheplatypus — CHANGES_REQUESTED** [review]

File: regulations/views/mixins.py (hunk 5 above)
Comment (cgodwin1) [reply regulations/views/mixins.py]: Resolved in b50e190

**Review by whytheplatypus — APPROVED** [review]

**Pull request merged**